ironfish/src/wallet/account.ts

```diff
@@ -205,8 +205,8 @@ export class Account {
       }
 
       const isRemovingTransaction = submittedSequence === null && blockHash === null
-      await this.bulkUpdateDecryptedNotes(transactionHash, decryptedNotes, tx)
-      await this.processTransactionSpends(transaction, isRemovingTransaction, tx)
+      await this.bulkUpdateDecryptedNotes(transactionHash, sequence, decryptedNotes, tx)
+      await this.processTransactionSpends(transaction, sequence, isRemovingTransaction, tx)
```

Comment:
I think in some cases, this is clearing out the sequence causing notes to get moved off chain. If you look at line 200, it only happens if the transaction should update but if the transaction hasn't changed and no sequence is passed then will this clear it?